src/main/java/world/bentobox/parkour/listeners/CourseRunnerListener.java

```diff
@@ -56,7 +57,7 @@ public CourseRunnerListener(Parkour addon) {
     public void onVisitorArrive(IslandEnterEvent e) {
         // Check if visitor
         User user = User.getInstance(e.getPlayerUUID());
-        if (user == null || !user.isOnline() || !addon.inWorld(e.getLocation())) {
+        if (user == null || !user.isOnline()) {
```

Comment:
i hope i haven't Chesterton's fenced this

Reply:
You don't need the null check anymore. The API improved since two years ago such that User.getInstance always returns nonNull. 

However, you do need to check this event applies to the Parkour world because the server can be running more than this game mode.

Reply:
e.getlocation seems to be the source location, not after the enter event, so with it doesnt trigger for entering the island from other worlds
overworld -> parkour
oneblock -> parkour
etc

the null check i didnt explicitly remove, since it was there before

Reply:
Good point on the location.